R/epi_shift_internal.R

```diff
@@ -0,0 +1,137 @@
+#' Create a shifted predictor
+#'
+#' `step_epi_shift` creates a *specification* of a recipe step that
+#'   will add new columns of shifted data. shifted data will
+#'   by default include NA values where the shift was induced.
+#'   These can be removed with [step_naomit()], or you may
+#'   specify an alternative filler value with the `default`
+#'   argument.
+#'
+#' @param shift A vector of integers. Each specified column will be
+#'  shifted for each value in the vector.
+#' @template step-return
+#'
+#' @details The step assumes that the data are already _in the proper sequential
+#'  order_ for shifting.
+#'
+#' @family row operation steps
+#' @export
+#' @rdname step_epi_ahead
+step_epi_shift <-
+  function(recipe,
+           ...,
+           role,
+           trained,
+           shift,
+           prefix,
+           default,
+           keys,
+           columns,
+           skip,
+           id) {
+    add_step(
+      recipe,
+      step_epi_shift_new(
+        terms = dplyr::enquos(...),
+        role = role,
+        trained = trained,
+        shift = shift,
+        prefix = prefix,
+        default = default,
+        keys = keys,
+        columns = columns,
+        skip = skip,
+        id = id
+      )
+    )
+  }
+
+step_epi_shift_new <-
```

Comment:
What does step_epi_shift_new do? Does it create a new shifted(lag or lead) column?

Reply:
the name step_epi_shift_new was a bit confusing to me

Reply:
I agree that this nesting is a bit bizarre, but it's seemingly the [standard for step creation](https://www.tidymodels.org/learn/develop/recipes/). We'll be making a lot of these over the coming weeks, so it's probably a good time to read over these instructions.

Reply:
@ChloeYou Yes, I just refactored the lag and ahead from the previous names of `step_epi_ahead_new` and `step_epi_lag_new`.